### PR TITLE
Update wrong GOOGLE_CHROME_BUILD usage

### PR DESCRIPTION
**Pull request by simonhong**
So far, we define explicitly GOOGLE_CHROME_BUILD in some specific source
files to use some code in that define guard.
However, some of them are replaced with
BUILDFLAG(GOOGLE_CHROME_BRANDING).
So, explicit define doesn't work after this change.

fix https://github.com/brave/brave-browser/issues/6497
fix brave/brave-browser#6496

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by simonhong** [comment]
@kjozwiak Confirmed that dmg from this PR plays well with netflix.

**Comment by simonhong** [comment]
@kjozwiak It's a known issue and the reason is reloading and installing are trigged in parallel when we click the install button. Maybe we already have issue for it.

**Comment by simonhong** [comment]
@mkarolin yeah, I agree. it seems better to free from upstream flags.

**Comment by kjozwiak** [comment]
> @kjozwiak [brave/brave-browser#4646](https://github.com/brave/brave-browser/issues/4646) is the reloading issue after installing widevine.

yup 👍Definitely a known issue and shouldn't block this particular fix. Would be nice to get fixed eventually but as mentioned above, refreshing fixes the problem.

File: patches/components-metrics-metrics_service_accessor.cc.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+index bddf51512c11d766921381d2fb4aed11b0ac7cd5..4312b8eabc3127ff88870f6a4aa5fe79e4b34546 100644
+--- a/components/metrics/metrics_service_accessor.cc
++++ b/components/metrics/metrics_service_accessor.cc
+@@ -26,7 +26,7 @@ bool IsMetricsReportingEnabledForOfficialBuild(PrefService* pref_service) {
+ // static
+ bool MetricsServiceAccessor::IsMetricsReportingEnabled(
+     PrefService* pref_service) {
+-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
++#if BUILDFLAG(GOOGLE_CHROME_BRANDING) || defined(BRAVE_CHROMIUM_BUILD)
+   return IsMetricsReportingEnabledForOfficialBuild(pref_service);
```
Comment (bridiver) [inline patches/components-metrics-metrics_service_accessor.cc.patch]: I don't understand this change, we always want metrics reporting to be false

File: patches/chrome-common-media-cdm_host_file_path.cc.patch [hunk 2]
```diff
@@ -1,7 +1,16 @@
...
-index a454dd3450ae06949ab38f2d854b7a9f811604ff..193d1a682aa2e07ac479ed819033ceba9610a0ff 100644
+index a454dd3450ae06949ab38f2d854b7a9f811604ff..1bb000892c5ce19238c72d53f46b5c3fb4faeaf8 100644
 --- a/chrome/common/media/cdm_host_file_path.cc
 +++ b/chrome/common/media/cdm_host_file_path.cc
+@@ -18,7 +18,7 @@
+ #include "chrome/common/chrome_constants.h"
+ #endif
+ 
+-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
++#if BUILDFLAG(GOOGLE_CHROME_BRANDING) || defined(BRAVE_CHROMIUM_BUILD)
```
Comment (bridiver) [inline patches/chrome-common-media-cdm_host_file_path.cc.patch]: GOOGLE_CHROME_BRANDING should be enabled by chromium_src override for this file

File: chromium_src/chrome/common/url_constants.cc [hunk 3]
```diff
@@ -45,22 +45,22 @@ const char kChromeReleaseNotesURL[] =
...
-#endif  // defined(GOOGLE_CHROME_BUILD)
+#endif  // BUILDFLAG(GOOGLE_CHROME_BRANDING)
 #else
     "https://support.brave.com/";
 #endif  // defined(OS_CHROMEOS)
 
 const char kChromeHelpViaMenuURL[] =
 #if defined(OS_CHROMEOS)
-#if defined(GOOGLE_CHROME_BUILD)
+#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
```
Comment (bridiver) [inline chromium_src/chrome/common/url_constants.cc]: why do we have these buildflags at all in our own file? That makes no sense.

File: browser/extensions/updater/brave_update_client_config.cc [hunk 4]
```diff
@@ -243,7 +244,7 @@ BraveUpdateClientConfig::GetProtocolHandlerFactory() const {
 
 update_client::RecoveryCRXElevator
 BraveUpdateClientConfig::GetRecoveryCRXElevator() const {
-#if defined(GOOGLE_CHROME_BUILD) && defined(OS_WIN)
+#if BUILDFLAG(GOOGLE_CHROME_BRANDING) && defined(OS_WIN)
```
Comment (bridiver) [inline browser/extensions/updater/brave_update_client_config.cc]: same as below, it makes no sense to have a GOOGLE_CHROME_ANYTHING buildflag in a brave file

File: browser/component_updater/brave_component_updater_configurator.cc [hunk 5]
```diff
@@ -185,7 +186,7 @@ BraveConfigurator::GetProtocolHandlerFactory() const {
 
 update_client::RecoveryCRXElevator BraveConfigurator::GetRecoveryCRXElevator()
     const {
-#if defined(GOOGLE_CHROME_BUILD) && defined(OS_WIN)
+#if BUILDFLAG(GOOGLE_CHROME_BRANDING) && defined(OS_WIN)
```
Comment (bridiver) [inline browser/component_updater/brave_component_updater_configurator.cc]: see below

File: patches/chrome-installer-util-google_update_settings.cc.patch [hunk 6]
```diff
@@ -14,23 +14,48 @@ index 441ed6d50d5cb4edcd665665fca7887f034ff6a0..709c7e5e24518902ae86ab1ef819dd9d
...
-+#endif
-+
  const GoogleUpdateSettings::UpdatePolicy
      GoogleUpdateSettings::kDefaultUpdatePolicy =
- #if BUILDFLAG(GOOGLE_CHROME_BRANDING)
-@@ -836,3 +845,7 @@ bool GoogleUpdateSettings::ReadExperimentLabels(
- 
-   return result == ERROR_SUCCESS;
+-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
++#if BUILDFLAG(GOOGLE_CHROME_BRANDING) || (defined(BRAVE_CHROMIUM_BUILD) && defined(OFFICIAL_BUILD))  // NOLINT
```
Comment (bridiver) [inline patches/chrome-installer-util-google_update_settings.cc.patch]: we do not want to enable google updates, I don't understand this change

File: browser/extensions/updater/brave_update_client_config.cc (hunk 4 above)
Comment (bridiver) [reply browser/extensions/updater/brave_update_client_config.cc]: should this just be OFFICIAL_BUILD or not exist at all?

File: chromium_src/chrome/common/url_constants.cc (hunk 3 above)
Comment (bridiver) [reply chromium_src/chrome/common/url_constants.cc]: it especially makes no sense because the url is the same

File: chromium_src/chrome/common/url_constants.cc [hunk 7]
```diff
@@ -45,22 +45,22 @@ const char kChromeReleaseNotesURL[] =
...
 #else
     "https://support.brave.com/";
-#endif  // defined(GOOGLE_CHROME_BUILD)
+#endif  // BUILDFLAG(GOOGLE_CHROME_BRANDING)
 #else
     "https://support.brave.com/";
 #endif  // defined(OS_CHROMEOS)
 
 const char kChromeHelpViaMenuURL[] =
 #if defined(OS_CHROMEOS)
```
Comment (bridiver) [inline chromium_src/chrome/common/url_constants.cc]: this whole section can be removed anyway, we don't need chromeos

File: chromium_src/chrome/common/url_constants.cc (hunk 3 above)
Comment (bridiver) [reply chromium_src/chrome/common/url_constants.cc]: apparently this file is auto generated so these changes won't work anyway

File: patches/chrome-installer-util-google_update_settings.cc.patch [hunk 8]
```diff
@@ -14,23 +14,48 @@ index 441ed6d50d5cb4edcd665665fca7887f034ff6a0..709c7e5e24518902ae86ab1ef819dd9d
...
+@@ -149,7 +153,7 @@ bool InitChannelInfo(bool system_install,
+   return channel_info->Initialize(key);
  }
-+
-+#if defined(BRAVE_CHROMIUM_BUILD) && defined(OFFICIAL_BUILD)
-+#undef GOOGLE_CHROME_BUILD
-+#endif
+ 
+-#if BUILDFLAG(GOOGLE_CHROME_BRANDING)
++#if BUILDFLAG(GOOGLE_CHROME_BRANDING) || (defined(BRAVE_CHROMIUM_BUILD) && defined(OFFICIAL_BUILD))  // NOLINT
```
Comment (bridiver) [inline patches/chrome-installer-util-google_update_settings.cc.patch]: this seems to actually do the opposite of what it did before and we're adding a bunch of patches for what was previous done in a few lines

File: patches/chrome-installer-util-google_update_settings.cc.patch (hunk 6 above)
Comment (mkarolin) [reply patches/chrome-installer-util-google_update_settings.cc.patch]: These aren't functional changes, we had it enabled before. 
The only difference is that before C78 this was under 
`#ifdef GOOGLE_CHROME_BUILD` 
and we used to define it just for the file (either via patch or via chromium_src).
Now, in C78 they are replacing `#ifdef GOOGLE_CHROME_BUILD` with
 `#if BUILDFLAG(GOOGLE_CHROME_BRANDING)`
If we wanted to make that enabled for the file, we'd have to expand their macros and do something like
`#if defined(BRAVE_CHROMIUM_BUILD) && defined(OFFICIAL_BUILD)
#define BUILDFLAG_INTERNAL_GOOGLE_CHROME_BRANDING 1
#endif
`
like we did before.
In other places we do patch with the || (defined(BRAVE_CHROMIUM_BUILD) && defined(OFFICIAL_BUILD)) and that gets us away from dependency on their future macro changes.

File: patches/chrome-common-media-cdm_host_file_path.cc.patch (hunk 2 above)
Comment (bridiver) [reply patches/chrome-common-media-cdm_host_file_path.cc.patch]: we already have a chromium_src file that does this and that needs to be changed from GOOGLE_CHROME_BUILD -> GOOGLE_CHROME_BRANDING

File: chromium_src/chrome/common/media/cdm_host_file_path.cc [hunk 9]
```diff
@@ -1,27 +0,0 @@
...
-#if defined(OS_MACOSX)
-#include "base/mac/bundle_locations.h"
-#include "chrome/common/buildflags.h"
-#include "chrome/common/chrome_constants.h"
-#endif
-
-// All above headers copied from original cdm_host_file_path.cc are included
-// to prevent below GOOGLE_CHROEM_BUILD affect them.
-
-#define GOOGLE_CHROME_BUILD
```
Comment (bridiver) [inline chromium_src/chrome/common/media/cdm_host_file_path.cc]: why did you delete this file instead of updating it to GOOGLE_CHROME_BRANDING? We don't want to add patches for things that don't need them

File: patches/chrome-installer-util-google_update_settings.cc.patch (hunk 6 above)
Comment (simonhong) [reply patches/chrome-installer-util-google_update_settings.cc.patch]: fyi, we use open source `omaha` for windows update.
omaha is opensource version of `google update`.
And we reuse most of `google update` integration logic in chromium.

File: chromium_src/chrome/common/url_constants.cc (hunk 3 above)
Comment (mkarolin) [reply chromium_src/chrome/common/url_constants.cc]: Removed changed to this file since they make no difference. If we want to clean it up it can be done as a separate effort.

File: patches/components-metrics-metrics_service_accessor.cc.patch (hunk 1 above)
Comment (mkarolin) [reply patches/components-metrics-metrics_service_accessor.cc.patch]: This has always been enabled for crash reporting.

File: patches/chrome-common-media-cdm_host_file_path.cc.patch (hunk 2 above)
Comment (mkarolin) [reply patches/chrome-common-media-cdm_host_file_path.cc.patch]: Fixed.

File: browser/extensions/updater/brave_update_client_config.cc (hunk 4 above)
Comment (mkarolin) [reply browser/extensions/updater/brave_update_client_config.cc]: Fixed.

File: browser/component_updater/brave_component_updater_configurator.cc (hunk 5 above)
Comment (mkarolin) [reply browser/component_updater/brave_component_updater_configurator.cc]: Fixed.

File: patches/chrome-installer-util-google_update_settings.cc.patch (hunk 8 above)
Comment (mkarolin) [reply patches/chrome-installer-util-google_update_settings.cc.patch]: Fixed.

File: chromium_src/chrome/common/media/cdm_host_file_path.cc (hunk 9 above)
Comment (mkarolin) [reply chromium_src/chrome/common/media/cdm_host_file_path.cc]: Fixed.

File: chromium_src/components/metrics/metrics_service_accessor.cc [hunk 10]
```diff
@@ -1,13 +1,38 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#if defined(BRAVE_CHROMIUM_BUILD)
-#define GOOGLE_CHROME_BUILD
-#endif
+#include "components/metrics/metrics_service_accessor.h"
```
Comment (bridiver) [inline chromium_src/components/metrics/metrics_service_accessor.cc]: this doesn't seem right, we only added GOOGLE_CHROME_BUILD for part of the file, I don't think we want it defined for the whole file

File: chromium_src/chrome/installer/util/google_update_settings.cc [hunk 11]
```diff
@@ -0,0 +1,51 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
```
Comment (bridiver) [inline chromium_src/chrome/installer/util/google_update_settings.cc]: this doesn't seem right, we only added GOOGLE_CHROME_BUILD for part of the file, I don't think we want it defined for the whole file

Comment (mkarolin) [reply chromium_src/chrome/installer/util/google_update_settings.cc]: The patch had it defined in 41 (after the headers and the other patch) and undefined in line 836 at the end of the file, as far as I can tell.

File: browser/component_updater/brave_component_updater_configurator.cc [hunk 12]
```diff
@@ -185,11 +185,7 @@ BraveConfigurator::GetProtocolHandlerFactory() const {
 
 update_client::RecoveryCRXElevator BraveConfigurator::GetRecoveryCRXElevator()
     const {
-#if defined(GOOGLE_CHROME_BUILD) && defined(OS_WIN)
```
Comment (bridiver) [inline browser/component_updater/brave_component_updater_configurator.cc]: why is this removed?

Comment (bridiver) [reply browser/component_updater/brave_component_updater_configurator.cc]: it seems like this was wrong from the start and should have be `IS_OFFICIAL_BUILD` for us?

Comment (mkarolin) [reply browser/component_updater/brave_component_updater_configurator.cc]: GOOGLE_CHROME_BUILD was never define for this file, as far as I can see, so the code inside was never used.

Comment (bridiver) [reply browser/component_updater/brave_component_updater_configurator.cc]: @bbondy is going to look into this for a potential follow-up

**Comment by bsclifton** [comment]
Waiting for CI, then can merge

**Comment by bsclifton** [comment]
Has a lint error... otherwise looks fine. Going to push a fix for that and verify lint passes, then will merge 👍

**Comment by bsclifton** [comment]
I don't think this is working as expected ☹️ When trying with an official build on Windows (Nightly version 0.73.17), I get the following when Widevine is installed and I try to view a video:
image

**Comment by bsclifton** [comment]
Tested again on Windows and macOS using https://github.com/brave/brave-browser/releases/tag/v0.73.19 and it worked great 👍 Please ignore the above (maybe the draft build was bad)
